modules/unsupported/mbstyle/src/main/java/org/geotools/mbstyle/function/MBFunction_toColor.java

```diff
@@ -0,0 +1,140 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2018, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.mbstyle.function;
+
+import org.geotools.filter.FunctionExpressionImpl;
+import org.geotools.filter.capability.FunctionNameImpl;
+import org.json.simple.JSONArray;
+import org.opengis.filter.capability.FunctionName;
+import org.opengis.filter.expression.Expression;
+import java.awt.*;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Takes an object as an argument and returns the color value if possible.
+ * Evaluates string in the formats of:
+ * "rgb(int, int, int)"
+ * "rgba(int, int, int, double)"
+ *
+ * And in arrays of 3 and 4 numbers for rgb and rgba colors:
+ * [int, int, int]
+ * [int, int, int, double]
+ *
+ * The integer values should be between 0-255, for red, green, and blue color values.
+ * The double value should be between 0-1, and is converted to a 0-255 alpha value.
+ */
+class MBFunction_toColor extends FunctionExpressionImpl {
```

Comment:
This "_" thing is due to the original FilterFunctions being autogenerated, please follow normal java class naming conventions if you are making functions.